site/app/controllers/course/CourseMaterialsController.php

```diff
@@ -355,229 +355,236 @@ public function ajaxUploadCourseMaterialsFiles(): JsonResponse {
 
-                        $zip = new \ZipArchive();
-                        $res = $zip->open($uploaded_files[1]["tmp_name"][$j]);
+                        $is_zip_file = false;
 
-                        if (!$res) {
-                            return JsonResponse::getErrorResponse("Failed to open zip archive");
+                        if (mime_content_type($uploaded_files[1]["tmp_name"][$j]) == "application/zip") {
+                            if (FileUtils::checkFileInZipName($uploaded_files[1]["tmp_name"][$j]) === false) {
+                                return JsonResponse::getErrorResponse("You may not use quotes, backslashes, or angle brackets in your filename for files inside " . $uploaded_files[1]['name'][$j] . ".");
+                            }
+                            $is_zip_file = true;
                         }
+                        //cannot check if there are duplicates inside zip file, will overwrite
+                        //it is convenient for bulk uploads
+                        if ($expand_zip == 'on' && $is_zip_file === true) {
+                            //get the file names inside the zip to write to the JSON file
 
-                        $entries = [];
-                        $disallowed_folders = [".svn", ".git", ".idea", "__macosx"];
-                        $disallowed_files = ['.ds_store'];
-                        for ($i = 0; $i < $zip->numFiles; $i++) {
-                            $entries[] = $zip->getNameIndex($i);
-                        }
-                        $entries = array_filter($entries, function ($entry) use ($disallowed_folders, $disallowed_files) {
-                            $name = strtolower($entry);
-                            foreach ($disallowed_folders as $folder) {
-                                if (Utils::startsWith($folder, $name)) {
-                                    return false;
-                                }
+                            $zip = new \ZipArchive();
+                            $res = $zip->open($uploaded_files[1]["tmp_name"][$j]);
+
+                            if (!$res) {
+                                return JsonResponse::getErrorResponse("Failed to open zip archive");
+                            }
+
+                            $entries = [];
+                            $disallowed_folders = [".svn", ".git", ".idea", "__macosx"];
+                            $disallowed_files = ['.ds_store'];
+                            for ($i = 0; $i < $zip->numFiles; $i++) {
+                                $entries[] = $zip->getNameIndex($i);
                             }
-                            if (substr($name, -1) !== '/') {
-                                foreach ($disallowed_files as $file) {
-                                    if (basename($name) === $file) {
+                            $entries = array_filter($entries, function ($entry) use ($disallowed_folders, $disallowed_files) {
```

Comment:
Should we alert the user if something is disallowed?

Reply:
I don't think so. These are things that should typically be not included in an upload. For example, people on a Mac would never see the .ds_store file but would get alerted every time they upload a zip. I think it is best to keep it how it is. Unless you have some suggestion here?